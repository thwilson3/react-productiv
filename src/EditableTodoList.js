import React from "react";
import EditableTodo from "./EditableTodo";

/** Show list of editable todos.
 *
 * Props:
 * - todos: array of [ todo, ... ]
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * TodoApp -> EditableTodoList -> [ EditableTodo, ... ]
 */

function EditableTodoList({ todos, update, remove }) {
  console.log(todos);
  function renderTodos() {
    return todos.map((t) => (
      <EditableTodo todo={t} update={update} remove={remove} />
    ));
  }
  return <div>{renderTodos()}</div>;
}

export default EditableTodoList;
