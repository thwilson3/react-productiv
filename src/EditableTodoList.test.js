import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { initialTodos } from "./App";
import { update, remove } from "./TodoApp";
import EditableTodoList from "./EditableTodoList";

describe("EditableTodoList", function() {
  it("renders without crashing", function () {
    render(
      <EditableTodoList todos={initialTodos} update={update} remove={remove} />
    );
  });

  it("renders list of todos", function() {
    const result = render(
      <EditableTodoList todos={initialTodos} update={update} remove={remove} />
    );

    for(let todo of initialTodos){
      expect(result.container.querySelector(`#todo-${todo.id}`)).toBeInTheDocument();
    }
  });

  it("works with no todos", function() {
    const result = render(
      <EditableTodoList todos={[]} update={update} remove={remove} />
    );
    expect(result.container.querySelector("div")).toBeEmptyDOMElement();
  });

  it("matches snapshot", function () {
    const { container } = render(
      <EditableTodoList todos={initialTodos} update={update} remove={remove} />
    );
    expect(container).toMatchSnapshot();
  });
})