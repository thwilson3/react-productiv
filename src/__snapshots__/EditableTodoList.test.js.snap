// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodoList matches snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-button-1"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="todo-1"
        >
          <div>
            <b>
              Code!
            </b>
             
            <small>
              (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              Write some code
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-button-2"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="todo-2"
        >
          <div>
            <b>
              Make dinner
            </b>
             
            <small>
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              Cook something healthy
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
            id="edit-button-3"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="todo-3"
        >
          <div>
            <b>
              Go to bed
            </b>
             
            <small>
              (priority: 
              3
              )
            </small>
          </div>
          <div>
            <small>
              In bed by 11:15
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
